packages/shared/src/lib/rank.ts

```diff
@@ -104,7 +104,15 @@ interface GetNextRankTextProps {
 }
 
 export const getRank = (rank: number): number => {
-  return rank === 0 ? rank : rank - 1;
+  if (rank < 0) {
+    return 0;
+  }
+
+  if (rank >= RANKS.length) {
+    return RANKS.length - 1;
+  }
+
+  return rank;
```

Comment:
I think you need to do the initial calculation first, it needs to do the -1 from what I recall. (as it's based on previous rank technically)

```suggestion
const calculatedRank = rank === 0 ? rank : rank - 1
  if (calculatedRank < 0) {
    return 0;
  }

  if (calculatedRank >= RANKS.length) {
    return RANKS.length - 1;
  }

  return calculatedRank;
```

Reply:
ah, great catch, updated 🙏